src/sdp/SDP.ts

```diff
@@ -658,4 +659,169 @@ export default class SDP {
+              prospectiveFmtpLine.startsWith(
+                `a=fmtp:${payloadMatch[1]} ${preference.codecCapability.sdpFmtpLine}`
+              )
+            ) {
+              codecMatches = true;
+              break;
+            }
+          }
+        } else {
+          // No 'fmtp' line, nothing else to check
+          codecMatches = true;
+        }
+
+        if (codecMatches) {
+          codecNamesToPayloadTypes.set(preference.codecName, payloadMatch[1]);
+          break;
+        }
+      }
+    });
+
+    // RFC 4566 5.14
+    // When a list of payload type numbers is given, this implies that all of these
+    // payload formats MAY be used in the session, but the first of these
+    // formats SHOULD be used as the default format for the session.
+
+    const payloadTypesToRemove: Set<string> = new Set(codecNamesToPayloadTypes.values());
+    // Remove payloads from the media line. m=video 9 UDP/+++ <payload> <payload> <payload> ...
+    const mline = lines[0].split(' ').filter((text: string) => !payloadTypesToRemove.has(text));
+    // Then splice them back in, in preferred order at the start of the list
+    const orderedPreferedPayloadTypes = Array.from(codecNamesToPayloadTypes.values()).sort(
+      (name1: string, name2: string) => {
+        const priority1 = preferences.findIndex(capability => {
+          return codecNamesToPayloadTypes.get(capability.codecName) === name1;
+        });
+        const priority2 = preferences.findIndex(capability => {
+          return codecNamesToPayloadTypes.get(capability.codecName) === name2;
+        });
+        return priority1 - priority2;
+      }
+    );
+    mline.splice(3, 0, ...orderedPreferedPayloadTypes.values());
+    lines[0] = mline.join(' ');
+
+    // Note that nothing in the RFCs require `a=rtpmap` lines to be reordered
+
+    return lines.join(SDP.CRLF) + SDP.CRLF;
+  }
+
+  // Returns undefined if there is no video send section or no codecs in the send section
+  highestPriorityVideoSendCodec(): VideoCodecCapability | undefined {
+    const srcSDP: string = this.sdp;
+    const sections = SDP.splitSections(srcSDP);
+    // Note `findActiveCameraSection` looks for `sendrecv` video sections so it
+    // works on both local and remote SDPs.
+    const cameraLineIndex: number = SDP.findActiveCameraSection(sections);
+    if (cameraLineIndex === -1) {
+      return undefined;
+    }
+
+    const lines = SDP.splitLines(sections[cameraLineIndex]);
+
+    // m=video 9 UDP/+++ <payload> <payload> <payload> ...
+    const mlineTokens = lines[0].split(' ');
+    if (mlineTokens.length < 4) {
+      return undefined;
+    }
+    const highestPriorityPayloadType = mlineTokens[3];
+    let highestPriorityCodecName: string = undefined;
+    let highestPriorityClockRate: string = undefined;
+    let highestPriorityFmtpLine: string = undefined;
+    for (const line of lines) {
+      if (highestPriorityCodecName !== undefined) {
+        continue; // Already found it
```

Comment:
Shouldn;t we just break after we find highest priority codec then?

Reply:
sure, originally had this in a `forEach` function :)